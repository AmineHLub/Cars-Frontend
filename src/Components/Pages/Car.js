const Car = () => (
  <div>Car</div>
);

export default Car;
