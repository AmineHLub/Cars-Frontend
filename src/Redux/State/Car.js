import Axios from 'axios';
import baseUrl from './baseUrl';

const GET_CARS = 'cars/GET_CARS';

export const fetchCars = () => async (dispatch) => {
  const response = await Axios.get(`${baseUrl}/cars`);
  dispatch({
    type: GET_CARS,
    payload: response.data,
  });
};

const initialState = [];

const carReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_CARS:
      return action.payload;
    default:
      return state;
  }
};

export default carReducer;
