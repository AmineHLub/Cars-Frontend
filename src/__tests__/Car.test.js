import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Car from '../components/Pages/Car';

const setPending = [];
const pendingReservations = [];

describe('Testing Car component', () => {
  test('renders a correct data', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router><Car pendingReservations={pendingReservations} setPending={setPending} /></Router>
      </Provider>,
    );
    expect(getByTestId('cars-container')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router><Car pendingReservations={pendingReservations} setPending={setPending} /></Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
