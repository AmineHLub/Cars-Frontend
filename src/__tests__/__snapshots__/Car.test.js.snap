// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Car component renders as expected 1`] = `
<div
  className="cars-container d-flex h-screen"
  data-testid="cars-container"
/>
`;
